### PR TITLE
Update useAnimatedKeyboard docs

### PR DESCRIPTION
**Pull request by maciekstosio**
## Summary

This PR updates the `useAnimatedKeyboard` docs to current doc pattern + add more detailed remarks discovered while doing #5851. It's pointing to #5851 and assumes changes from that PR, so **should be merged after**. 

## Test plan

```
cd docs
yarn
yarn start
```

and go to useAnimatedKeyboard section

### CODE REVIEW

File: docs/docs/device/useAnimatedKeyboard.mdx [hunk 1]
```diff
@@ -2,72 +2,112 @@
...
+  }
+/>
 
-#### `AnimatedKeyboardOptions: [object]`
+## Remarks
 
-Properties:
+- On Android, make sure to set `android:windowSoftInputMode` in your `AndroidManifest.xml` to `adjustResize`. Then, using the `useAnimatedKeyboard` hook disables
+  the default Android behavior (resizing the view to accommodate keyboard) in the whole app. Using values from `useAnimatedKeyboard` hook you can handle the keyboard yourself. Unmounting all components that use `useAnimatedKeyboard` hook brings back the default Android behavior.
+- On Android using the `useAnimatedKeyboard` hook expands root view to full screen (immersive mode) and takes control over insets management. It applies the top and bottom margins according to the insets when `isStatusBarTranslucentAndroid` is false and sets top margin to 0 and bottom padding according to the navigation insets when `isStatusBarTranslucentAndroid` is true.
```
Comment (tomekzaw) [inline docs/docs/device/useAnimatedKeyboard.mdx:101]:
```suggestion
- On Android, using the `useAnimatedKeyboard` hook expands root view to full screen (immersive mode) and takes control over insets management. It applies the top and bottom margins according to the insets when `isStatusBarTranslucentAndroid` is false and sets top margin to 0 and bottom padding according to the navigation insets when `isStatusBarTranslucentAndroid` is true.
```

File: docs/docs/device/useAnimatedKeyboard.mdx [hunk 2]
```diff
@@ -2,72 +2,112 @@
...
 
-#### `AnimatedKeyboardOptions: [object]`
+## Remarks
 
-Properties:
+- On Android, make sure to set `android:windowSoftInputMode` in your `AndroidManifest.xml` to `adjustResize`. Then, using the `useAnimatedKeyboard` hook disables
+  the default Android behavior (resizing the view to accommodate keyboard) in the whole app. Using values from `useAnimatedKeyboard` hook you can handle the keyboard yourself. Unmounting all components that use `useAnimatedKeyboard` hook brings back the default Android behavior.
+- On Android, using the `useAnimatedKeyboard` hook expands root view to full screen (immersive mode) and takes control over insets management. It applies the top and bottom margins according to the insets when `isStatusBarTranslucentAndroid` is false and sets top margin to 0 and bottom padding according to the navigation insets when `isStatusBarTranslucentAndroid` is true.
+- On Android, when using navigation with native header, `isStatusBarTranslucentAndroid` doesn't affect the top inset.
+- On Android SDK < 30, when status bar is hidden, the keyboard reverts to the default Android behavior.
```
Comment (kacperkapusciak) [inline docs/docs/device/useAnimatedKeyboard.mdx:103]: Please add an empty new line between each points for some breathing space

File: docs/docs/device/useAnimatedKeyboard.mdx [hunk 3]
```diff
@@ -2,72 +2,112 @@
...
+  }
+/>
 
-#### `AnimatedKeyboardOptions: [object]`
+## Remarks
 
-Properties:
+- On Android, make sure to set `android:windowSoftInputMode` in your `AndroidManifest.xml` to `adjustResize`. Then, using the `useAnimatedKeyboard` hook disables
+  the default Android behavior (resizing the view to accommodate keyboard) in the whole app. Using values from `useAnimatedKeyboard` hook you can handle the keyboard yourself. Unmounting all components that use `useAnimatedKeyboard` hook brings back the default Android behavior.
+- On Android, using the `useAnimatedKeyboard` hook expands root view to full screen (immersive mode) and takes control over insets management. It applies the top and bottom margins according to the insets when `isStatusBarTranslucentAndroid` is false and sets top margin to 0 and bottom padding according to the navigation insets when `isStatusBarTranslucentAndroid` is true.
```
Comment (kacperkapusciak) [inline docs/docs/device/useAnimatedKeyboard.mdx:101]:
```suggestion
- On Android, using the `useAnimatedKeyboard` hook expands root view to full screen ([immersive mode](https://developer.android.com/develop/ui/views/layout/immersive)) and takes control over insets management. It applies the top and bottom margins according to the insets when `isStatusBarTranslucentAndroid` is `false` and sets top margin to `0` and bottom padding according to the navigation insets when `isStatusBarTranslucentAndroid` is `true`.
```

Lets add a link to the official android documentation explaining what immersive mode is.

This point could be split into subpoints for easier text skimming eg.
- On Android...
  - when `isStatusBarTranslucentAndroid` is set `true` this happens
  - when `isStatusBarTranslucentAndroid` is set to false that happens

**Review by kacperkapusciak — APPROVED** [review]
Great work! Please address the comments before merging but other than that lgtm!

**Pull request merged**